packages/keyring-controller/src/KeyringController.ts

```diff
@@ -479,18 +509,18 @@ export class KeyringController extends BaseController<
    * @param address - Address of the account to remove.
    * @returns Promise resolving current state when this account removal completes.
    */
-  async removeAccount(address: string): Promise<KeyringMemState> {
+  async removeAccount(address: string): Promise<KeyringControllerState> {
```

Comment:
I am conflicted about what we should do with this "return the entire keyring state" methods.

That pattern was useful because the controller state only contained the persisted state, so this was the only way to get the in-memory state. But now that the in-memory is available as controller state as well, I don't think this serves a purpose anymore.

Perhaps we should update all of these methods to stop returning the state?

Reply:
The affected methods by this change would be:
* `addNewAccount`
* `addNewAccountWithoutUpdate`
* `createNewVaultAndRestore`
* `createNewVaultAndKeychain`
* `importAccountWithStrategy`
* `removeAccount`
* `setLocked` (maybe?)

I would do this in a different PR as we should do some considerations on which return we should leave in some cases (e.g. `addNewAccount` should still return created address, `createNewVaultAndKeychain` returns the state but that's actually what is being created by that method so might make sense).

We could also spare the Memstore sync for the future, but I think that being able to access the memstore from the KeyringController is [the main reason](https://github.com/MetaMask/core/issues/1103#issuecomment-1509264365) we need this BaseControllerV2 migration.

So if that make sense to you, I would leave this PR as is now, and open another issue to address the `KeyringMemState` returns

Reply:
Hmm, OK.

Perhaps instead we could leave the return type for these methods unchanged by excluding the `vault` from the returned state? That would further reduce the scope of changes, and avoid the risk of the vault being logged somewhere by accident or something.

Reply:
Ahh good point. Hmm. Perhaps we can exclude those as well, to keep the old state shape for now since they aren't useful for anyone using this return value.

The encryption key is indeed sensitive, though the salt is not.

Reply:
I removed `vault`, `encryptionKey` and `encryptionVault` from return statements, creating a new type `KeyringControllerMemState`; it probably isn't an accurate naming but it is closer to how it was called before.
I had to edit a couple of tests that were checking encryption values, checking them in the `controller.state` instead of the function return (as we don't have them there anymore)